From 1633f2962e195eb2a013072e694a2e1c701613a8 Mon Sep 17 00:00:00 2001
From: 0xFelix <evostar@gmx.de>
Date: Tue, 12 Dec 2017 13:24:34 +0100
Subject: [PATCH] Remove usage of deprecated avfiltergraph.h header

avfiltergraph.h was replaced by avfilter.h in libavfilter version
3.8.0+ so only include it when the used libavfilter version
is older than 3.8.0
---
 src/QtAV/private/AVCompat.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/QtAV/private/AVCompat.h b/src/QtAV/private/AVCompat.h
index 6c38596d1..944cfd7de 100644
--- a/src/QtAV/private/AVCompat.h
+++ b/src/QtAV/private/AVCompat.h
@@ -59,6 +59,7 @@ extern "C"
 #include <libavutil/parseutils.h>
 #include <libavutil/pixdesc.h>
 #include <libavutil/avstring.h>
+#include <libavfilter/version.h>
 
 #if !FFMPEG_MODULE_CHECK(LIBAVUTIL, 51, 73, 101)
 #include <libavutil/channel_layout.h>
@@ -79,8 +80,11 @@ extern "C"
 #endif //QTAV_HAVE(AVRESAMPLE)
 
 #if QTAV_HAVE(AVFILTER)
+#if LIBAVFILTER_VERSION_INT < AV_VERSION_INT(3,8,0)
 #include <libavfilter/avfiltergraph.h> /*code is here for old version*/
+#else
 #include <libavfilter/avfilter.h>
+#endif
 #include <libavfilter/buffersink.h>
 #include <libavfilter/buffersrc.h>
 #endif //QTAV_HAVE(AVFILTER)
